### PR TITLE
Expanded the documentation about the stable and unstable servers

### LINKED ISSUE #9718: Expand our documentation of stable/unstable build servers
We had a great discussion among @publiclab/soc team today about the stable/unstable servers, and realized we could do a lot to better explain them, beyond technically how to use them but to include their purpose, our norms and habits around them, and tips!

Here are some notes we took; it'd be great to have them added to the READMEs: (https://github.com/publiclab/plots2/blob/main/doc/TESTING.md#testing-your-work-with-unstable)

****

```
### Stable server

Purpose of https://stable.publiclab.org

Info:

- `main` branch auto-builds at https://stable.publiclab.org
- "Pre-production"
- Last manual checks before it publishes live
- Last chance to find bugs!
- You can try things out on here - it’s a copy of the database, so you can post “example” or “test” posts and nobody will see on the live site
- No emails are sent
- Build process: merge to main => auto-triggers build of stable.publiclab.org => push to stable branch => publish on live site at publiclab.org
- Passwords are all replaced with "publiclab" for security reasons

### Unstable server

Anyone in the org can push a branch to it - it simulates the live site with the full database
https://github.com/publiclab/plots2/blob/main/doc/TESTING.md#testing-your-work-with-unstable (documentation)

Commands:

Push to unstable branch of publiclab/plots2 to trigger a build:  $ git push -f git@github.com:publiclab/plots2 HEAD:unstable

Monitor all builds at https://jenkins.laboratoriopublico.org/
```

If someone would like to try cleaning these notes up and making them more readable and friendly, I'd super appreciate it!

### PR DESCRIPTION
**Pull request by anirudhprabhakaran3**
Fixes #9718  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [X] PR is descriptively titled 📑 and links the original issue above 🔗
* [X] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [X] code is in uniquely-named feature branch and has no merge conflicts 📁
* [X] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by anirudhprabhakaran3** [comment]
Why are so many tests failing, even when I'm just making changes to docs?

**Comment by RuthNjeri** [comment]
Thanks for opening this up @anirudhprabhakaran3, I can see that you have not included all the information from the issue, for example, the purpose of stable.org, where you can monitor all builds, etc...

**Comment by anirudhprabhakaran3** [comment]
Aah yes I seem to have missed a few points. I'll add those asap. 

For the builds, I've made a different subsection at the end about build status and added the jenkins link. Is that what you are referring to?

File: doc/TESTING.md [hunk 1]
```diff
@@ -161,17 +161,28 @@ It is a staging server that includes a copy of the production database and is in
...
-**Email testing:** _unstable_ server can't send emails so Email related features can't be tested.  
+**Environment:** Test environment<br>
+**Database:** MySQL<br>
+**Data:** Data present at https://unstable.publiclab.org is updated occasionally using the Production database, so it has good amount of data for testing. <br>
+**Email testing:** _unstable_ server can't send emails so Email related features can't be tested.  <br>
 
 ### Pushing on `unstable`
 
 1. Drop a message in the [Gitter chatroom](https://gitter.im/publiclab/publiclab) that you are pushing to _unstable_, as other members also use it for testing their work so you don't want to shock them by pushing your work over theirs. After dropping the message, wait for 5 mins, and if no one mentions that they are using the branch, then go to step 2.
-2. Push using: `git push -f https://github.com/publiclab/plots2.git HEAD:unstable`.<br><br>**Note:** If you get this message - `remote: Permission to publiclab/plots2.git denied to <username>` then you don't have permission to push to _unstable_ branch.
```
Comment (pydevsg) [inline doc/TESTING.md:172]: You don't need to make any change here @anirudhprabhakaran3 .

**Review by pydevsg — COMMENTED** [review]
Thanks for writing this @anirudhprabhakaran3 . 😄 
Make a small change as asked , rest looks great to me .

**Comment by RuthNjeri** [comment]
> Aah yes I seem to have missed a few points. I'll add those asap.
> 
> For the builds, I've made a different subsection at the end about build status and added the jenkins link. Is that what you are referring to?

Yes, I do see the Jenkins link but the purpose of the stable site from the issue is not present yet.

File: doc/TESTING.md (hunk 1 above)
Comment (anirudhprabhakaran3) [reply doc/TESTING.md]: I've made the change. Thanks!

**Comment by RuthNjeri** [comment]
Hi @anirudhprabhakaran3, are you having trouble with this? We are happy to help if you need any assistance.

File: doc/TESTING.md [hunk 2]
```diff
@@ -174,4 +174,21 @@ But, don't worry, feel free to ping @publiclab/maintainers on your open PR to re
 3. See build progress at https://jenkins.laboratoriopublico.org/job/Plots-Unstable/. It takes 8-15 mins for build to complete. 
 4. Test website: https://unstable.publiclab.org/. 
 
-Once your PR is merged into plots2/main, code is only deployed to the live site [publiclab.org](publiclab.org) after some time by our maintainers. (They often send a message in our chatroom to notify contributors of this update). Fortunately, you can see your changes implemented immediately at https://stable.publiclab.org, which builds from the main branch. This is useful for testing your work again after a merge.
+
+### Stable Server
+
+You've tested your code on the unstable server. Now, once your PR is merged into plots2/main, it auto-triggers the build of the [stable version](https://stable.publiclab.org) of the website. This is the "pre-production" server.
```
Comment (jywarren) [inline doc/TESTING.md:180]:
```suggestion
Whether or not you've tested your code on the unstable server, once your PR is merged into `plots2/main`, it auto-triggers the build of the [stable version](https://stable.publiclab.org) of the website. This is the "pre-production" server and the last chance to test out any changes before they're published to the live site. 
```

File: doc/TESTING.md [hunk 3]
```diff
@@ -174,4 +174,21 @@ But, don't worry, feel free to ping @publiclab/maintainers on your open PR to re
...
+
+The stable server also uses a copy of the production database, so you can make 'example' and 'test' posts and no one will see it on the live website. However, emails aren't sent, which means you won't be able to test them. All passwords are also replaced with "publiclab" for security reasons.
+
+The exact workflow is this:<br>
+1. Merge to main <br>
+2. Auto-trigger build of stable.publiclab.org <br>
+3. Push to the stable branch <br>
+4. Publish on the live site at publiclab.org <br>
+
+After some time, one of the maintainers deploys the code to the [live site](https://publiclab.org). Usually, they send a message in the chatroom to notify other contributors of the update. Your code is now live!
```
Comment (jywarren) [inline doc/TESTING.md:191]:
```suggestion
After some time (usually a week or two), one of the maintainers deploys the code to the [live site](https://publiclab.org). Usually, they send a message in the [chatroom](http://publiclab.org/chat) to notify other contributors of the update. Your code is now live!
```

**Review by jywarren — APPROVED** [review]
Looks great! I made some small changes. We can make more refinements later! Thanks!

**Comment by jywarren** [comment]
Excellent, than kyou!

**Pull request merged**